### PR TITLE
A bunch of changes

### PR DESCRIPTION
**Pull request by alagos**
Change details:

- Adding lambdas in scopes and relationships (mostly fixed in PR https://github.com/lesliev/oscurrency/pull/4 )
- Some lines had `where(status: 0)`, which it was failing. Changed to `where(memberships: { status: 0 })` as status is in memberships table.
- Removed date fields from fixtures because it was causing tests to fail (and because there shouldn't be a real reason to set `created_at`/`updated_at` fields in 99.99% of tests in general).
- Upgrading to last stable rails 4.0.x version (In the hope this solved some issues. It didn't, but didn't make fail other things, so all good).
- Several deprecation warnings fixed.

Additional testing notes:
I was having [this issue](https://github.com/rails/rails/issues/28780) when I run `rspec`, which it solved downgrading PostgreSQL to 9.x. [This is the guide](https://medium.com/keeping-code/running-multiple-postgresql-versions-simultaneously-on-macos-linux-90b3d7e08ffd) I followed to use multiple psql versions.
TL;DR (mostly for Debian-based distros)
```bash
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -\n
sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" >> /etc/apt/sources.list.d/pgdg.list'
sudo apt-get update
sudo apt install postgresql-9.6
pg_createcluster 9.6 main
pg_lsclusters # Check 9.6 port and add it to config/database.yml
sudo su - postgres
psql --cluster 9.6/main
CREATE ROLE youruser WITH SUPERUSER LOGIN PASSWORD 'yourpassword';
\q
exit
RAILS_ENV=test rake db:setup
```
At least now there are a few tests passing, but most of them are failing with [`NoMethodError: undefined method '[]' for nil:NilClass`](https://stackoverflow.com/questions/23345304/undefined-method-for-nilnilclass-form), which is because we're using `protected_attributes` gem, so it should be fixed when finally moved to strong parameters.

### CODE REVIEW

**Comment by lesliev** [comment]
Thanks so much Alter! I'll get on this soon!

**Review by lesliev — APPROVED** [review]
Awesome!
